### PR TITLE
Enable vulkan debug markers in release builds

### LINKED ISSUE #3482: Debug markers in release build
Right now, debug markers only available if the debug_assertions feature is set. This is true at least on the [Vulkan](https://github.com/gfx-rs/gfx/blob/f09b062c5159b4a9dd8879dff3c0a3ac1bd81ea7/src/backend/vulkan/src/lib.rs#L440) and [DirectX 12](https://github.com/gfx-rs/gfx/blob/f09b062c5159b4a9dd8879dff3c0a3ac1bd81ea7/src/backend/dx12/src/lib.rs#L751) backends, but I'd imagine the others as well. It would be nice if there were a feature flag or other way to enable debug markers in release builds as well.

### PR DESCRIPTION
**Pull request by fintelia**
Fixes #3482 (at least for the Vulkan backend)
PR checklist:
- [x] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:

### CODE REVIEW

File: src/backend/vulkan/src/lib.rs [hunk 1]
```diff
@@ -437,7 +433,6 @@ impl hal::Instance<Backend> for Instance {
                 })
             });
 
-        #[cfg(debug_assertions)]
         let debug_messenger = {
             // make sure VK_EXT_debug_utils is available
             if instance_extensions.iter().any(|props| unsafe {
```
Comment (kvark) [inline src/backend/vulkan/src/lib.rs:438]: there is a change missing here. We don't want to request the `VK_EXT_debug_utils` or `DebugReport::name()` if running in release. So these conditions should look like:
```rust
if cfg!(debug_assertions) && instance_extensions.iter().any(...) {
  ...
} else if cfg!(debug_assertions) && instance_extensions.iter().any(..) {
  ...
} else {
  None
}
```

**Review by kvark — CHANGES_REQUESTED** [review]

File: src/backend/vulkan/src/lib.rs (hunk 1 above)
Comment (fintelia) [reply src/backend/vulkan/src/lib.rs:438]: If we don't request debug utils, then [insert_debug_marker will be a no-op.](https://github.com/gfx-rs/gfx/blob/e99e71d0022f72fe4f558ce9f528e258131af2a5/src/backend/vulkan/src/command.rs#L1121)

Comment (kvark) [reply src/backend/vulkan/src/lib.rs:438]: Oh snap, you are right. In this case, we should at least gate the `DebugReport::name()` by `debug_assertions`?

**Review by kvark — APPROVED** [review]
Thank you!
bors r+

**Comment by kvark** [comment]
Uh, Rust nightly bug :(
Could you comment out the macOS nightly build from bors.toml for now?

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors retry

**Pull request merged**